src/lava/lib/optimization/solvers/generic/hierarchical_processes.py

```diff
@@ -357,3 +357,118 @@ def __init__(
         self.cost_diagonal = Var(shape=shape, init=cost_diagonal)
         self.assignment = Var(shape=shape, init=False)
         self.min_cost = Var(shape=shape, init=False)
+
+
+class NEBMSimulatedAnnealingAbstract(AbstractProcess):
+    r"""Event-driven stochastic discrete dynamical system with two outputs.
+
+    The main output is intended as input to other dynamical systems on
+    the network, whilst the second output is to transfer local information to be
+    integrated by an auxiliary dynamical system or circuit.
+
+    Attributes
+    ----------
+    added_input: InPort
+        The addition of all inputs (per dynamical system) at this
+        timestep will be received by this port.
+    replace_assignment: InPort
+        Todo: deprecate
+    messages: OutPort
+        The payload to be sent to other dynamical systems when firing.
+    local_cost: OutPort
+        the cost component corresponding to this dynamical system, i.e.,
+        c_i = sum_j{Q_{ij} \cdot x_i}  will be sent through this port. The cost
+        integrator will then complete the cost computation  by adding all
+        contributions, i.e., x^T \cdot Q \cdot x = sum_i{c_i}.
+
+    """
+
+    def __init__(
+            self,
+            *,
+            max_temperature: int = 10,
+            min_temperature: int = 0,
+            delta_temperature: int = 1,
+            steps_per_temperature: int = 100,
+            refract_scaling: int = 2,

```

Comment:
The refractory values should be in the range ~ 0-20. Thus, we need refract_scaling to be rather 
refract_scaling = 14
?